pkg/cloud-init/cloud-init_test.go

```diff
@@ -204,6 +204,17 @@ var _ = Describe("CloudInit", func() {
 					verifyCloudInitIso(cloudInitData)
 				})
 			})
+			Context("with cloudInitNoCloud userData volume source and networkData", func() {
```

Comment:
Testing for anything but the presence might be tough. The get resolv function isn't mocked out and it would be test environment dependent. I'll add some basic testing for now.